dbms/src/Columns/ColumnString.cpp

```diff
@@ -1348,42 +1145,20 @@ void ColumnString::deserializeAndInsertFromPosForColumnArrayImpl(
 
     size_t size = pos.size();
     size_t char_size = chars.size();
-    if constexpr (compare_semantics)
```

Comment:
In `serializeToPosForColumnArrayImpl`, there is a branch for `need_decode_collator`. But in `deserializeAndInsertFromPosForColumnArrayImpl`, this branch is removed?